File: Java/Subarray-Sum.java
public class Solution {
    /**
     * @param nums: A list of integers
     * @return: A list of integers includes the index of the first number
     *          and the index of the last number
     */
    public ArrayList<Integer> subarraySum(int[] nums) {
        // write your code here
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
        ArrayList<Integer> res = new ArrayList<Integer>();
        hash.put(0, -1);
        int sum = 0;
        for (int i = 0; i < nums.length; i ++) {
            sum += nums[i];
            if (hash.containsKey(sum)) {
                res.add(hash.get(sum) + 1);
                res.add(i);
                return res;
            } else {
                hash.put(sum, i);
            }
        }
        return res;
    }
}

File: Python/Subarray-Sum.py
class Solution:
    """
    @param nums: A list of integers
    @return: A list of integers includes the index of the first number
             and the index of the last number
    """
    def subarraySum(self, nums):
        # write your code here
        dict = {}
        dict[0] = -1
        res = []
        cur = 0
        for i in range(len(nums)):
            cur += nums[i]
            if cur in dict:
                res.append(dict[cur] + 1)
                res.append(i)
                return res
            dict[cur] = i
        return res